stdlib/string.gr

```diff
@@ -2338,3 +2338,45 @@ provide let toAsciiUppercase = string => {
   }
   implode(chars)
 }
+
+/**
+ * Produces a new string by repeating a given substring a given number of times.
+ *
+ * @param string: The string to repeat
+ * @param count: The number of times to repeat, negative values reverse the substring
+ * @returns A string containing the repeated input string
+ *
+ * @throws InvalidArgument(String): When the `count` is not an integer
+ *
+ * @example assert String.repeat("=", 5) == "====="
+ * @example assert String.repeat("ab", -5) == "bababababa"
+ * @example assert String.repeat(".", 0) == ""
+ *
+ * @since v0.6.7
+ */
+@unsafe
+provide let repeat = (string, count) => {
+  use WasmI32.{ (+), (*), (<), (&), (!=) }
+  if ((WasmI32.fromGrain(count) & 1n) != 1n) {
+    throw InvalidArgument("Invalid count value")
+  }
+  let mut rawCount = untagSimpleNumber(count)
+  if (WasmI32.eqz(rawCount)) return ""
+  let string = if (rawCount < 0n) {
+    rawCount *= -1n
+    reverse(string)
```

Comment:
Do other languages do this? This seems like really surprising behavior—I'd probably just want an error.

Reply:
So checking a few other languages,
``js
"=".repeat(-5)
```
results in a range error.
```py
"=" * -5
```
results in an empty string


I decided to make it flip the string so we handled the behaviour instead of throwing an exception, but I think I am going to make it throw an exception following the Javascript behaviour.